src/fee-calculator.ts

```diff
@@ -40,8 +46,11 @@ export class FeeCalculator {
       }
     }
 
-    // Only write the report if we have data for at least one distributor
-    if (Object.keys(feeReport.distributors).length > 0) {
+    // Write the report if we have data OR if a specific distributor was requested
+    if (
+      Object.keys(feeReport.distributors).length > 0 ||
+      distributorAddress !== undefined
```

Comment:
why do we check this? just because we supply a distributor address doesnt mean we'll fine it in the file existing data does it? we should throw an error if the distributor provided does not exist in the data from out file manager